src/Tes.ApiClients/CachingRetryHandler.cs

```diff
@@ -43,12 +44,13 @@ public CachingRetryHandler(IMemoryCache appCache, IOptions<RetryPolicyOptions> r
         /// <param name="cacheKey"></param>
         /// <param name="action">Action to execute</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> for controlling the lifetime of the asynchronous operation.</param>
+        /// <param name="context"></param>
         /// <returns></returns>
-        public virtual async Task<TResult> ExecuteWithRetryAndCachingAsync<TResult>(string cacheKey, Func<CancellationToken, Task<TResult>> action, CancellationToken cancellationToken)
+        public virtual async Task<TResult> ExecuteWithRetryAndCachingAsync<TResult>(string cacheKey, Func<CancellationToken, Task<TResult>> action, CancellationToken cancellationToken, Context? context = default)
```

Comment:
I think exposing the Polly context to the caller is unnecessary, as it does not provide much benefit, and it leads to many changes that we have to maintain for every API call. We can log the retries without passing the context to the caller, as the runner already does with its policy - [see this](https://github.com/microsoft/ga4gh-tes/blob/d470a118a0c448bc55c869c8809324de7468076f/src/Tes.Runner/Transfer/BlobApiHttpUtils.cs#L350). I realize that this runner's retry approach relies on the exception stack to identify the failed method and might produce a verbose log output, unlike the PR approach. However, I prefer this simpler option.

Reply:
I thought of having this be opt-in and flexible, and rather than having to craft unique policies all over the codebase, reusing the ones already provided in `RetryHandler` seemed more maintainable to my mind. It allows for flexibility in extending the policies as well, without restricting usage to only the generally available policies.

What I didn't do was provide a default retry log in-place.

Reply:
I just took it a step closer: it's still opt-in, but there's no longer any exposing of `Polly.Context`s. It does reduce the line count of the changes overall.

I'm toying with an alternative: making the current `Tes.ApiClients.RetryHandler` into an opinionated retry policy builder based on `RetryOptions` (and extend the builder with constructs from `CachedRetryHandler`). That would pull together all the information needed for the logs without hitting every single line of code where a retry needs to be handled.

Reply:
Sounds good, I agree that moving this functionality to the `CachingRetryHandler` is a better approach.  Also, if making it, opt-in requires many changes, that is something you can defer as in practicality I'd always want to see these retries for troubleshooting.